lib/cocoapods/installer/user_project_integrator/target_integrator.rb

```diff
@@ -433,6 +433,87 @@ def embed_frameworks_output_paths(framework_paths, xcframeworks)
             end
             paths + xcframework_paths
           end
+
+          # Updates a projects native targets to include on demand resources specified by the supplied parameters.
+          # Note that currently, only app level targets are allowed to include on demand resources.
+          #
+          # @param  [Sandbox] sandbox
+          #         The sandbox to use for calculating ODR file references.
+          #
+          # @param  [Xcodeproj::Project] project
+          #         The project to update known asset tags as well as add the ODR group.
+          #
+          # @param  [Xcodeproj::PBXNativeTarget, Array<Xcodeproj::PBXNativeTarget>] native_targets
+          #         The native targets to integrate on demand resources into.
+          #
+          # @param  [Sandbox::FileAccessor, Array<Sandbox::FileAccessor>] file_accessors
+          #         The file accessors that that provide the ODRs to integrate.
+          #
+          # @param  [Xcodeproj::PBXGroup] parent_odr_group
+          #         The group to use as the parent to add ODR file references into.
+          #
+          # @param  [String] target_odr_group_name
+          #         The name to use for the group created that contains the ODR file references.
+          #
+          # @return [void]
+          #
+          def add_on_demand_resources(sandbox, project, native_targets, file_accessors, parent_odr_group,
```

Comment:
this became a "helper" class method as the others. Helps do the job easily without repeating stuff in both `target_integrator` and `pod_target_integrator`.